pytket/binders/circuit/boxes.cpp

```diff
@@ -186,14 +186,57 @@ void init_boxes(py::module &m) {
             boost::bimap<Qubit, unsigned> bmap;
             for (const auto &pair : q_ind) {
               bmap.insert({pair.first, pair.second});
+              if (pair.second >= n_qb) {
```

Comment:
I wanted to not have this in the c++ layer because it would increasing the runtime quiet a lot to check every box construction in this much details. Happy to change that if you think that is not a good reason to have this check in python.

Reply:
I think it's better that it goes in the core code. It doesn't look very expensive to me, and we should guarantee that these objects are valid on construction.